numpy/lib/_shape_base_impl.py

```diff
@@ -403,17 +406,8 @@ def apply_along_axis(func1d, axis, arr, *args, **kwargs):
     for ind in inds:
         buff[ind] = asanyarray(func1d(inarr_view[ind], *args, **kwargs))
 
-    if not isinstance(res, matrix):
-        # wrap the array, to preserve subclasses
-        buff = res.__array_wrap__(buff)
-
-        # finally, rotate the inserted axes back to where they belong
-        return transpose(buff, buff_permute)
-
-    else:
-        # matrices have to be transposed first, because they collapse dimensions!
-        out_arr = transpose(buff, buff_permute)
-        return res.__array_wrap__(out_arr)
+    res = transpose(buff, buff_permute)
```

Comment:
For subclasses other than `matrix`, this would be an API change if `.transpose()` is overridden. That seems rather unlikely (and wrong-headed), so I'm fine with making the change regardless...